src/node/mod.rs

```diff
@@ -653,10 +653,19 @@ impl Node {
                 _ => unreachable!(),
             },
             Stage::Approved(stage) => match msg.variant() {
-                Variant::NeighbourInfo { elders_info, .. } => {
+                Variant::NeighbourInfo {
+                    elders_info,
+                    section_key,
+                    nonce,
+                } => {
                     msg.dst().check_is_section()?;
-                    let src_key = *msg.proof_chain_last_key()?;
-                    stage.handle_neighbour_info(&mut self.core, elders_info.clone(), src_key)?;
+                    stage.handle_neighbour_info(
+                        &mut self.core,
+                        elders_info.value.clone(),
+                        *section_key,
```

Comment:
So I think this can be replaced with something like `msg.proof_chain_last_key()?` and then we wouldn't need to include the `section_key` in the `NeighbourInfo` variant.